nipoppy/workflow/bids_conv/run_dcm2bids.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import glob
+import shutil
+import subprocess
+from joblib import Parallel, delayed
+from pathlib import Path
+from typing import Iterable
+
+import numpy as np
+from bids.layout import parse_file_entities
+
+import nipoppy.workflow.catalog as catalog
+import nipoppy.workflow.logger as my_logger
+from nipoppy.workflow.utils import (
+    COL_CONV_STATUS,
+    COL_DICOM_ID,
+    DNAME_BACKUPS_DOUGHNUT,
+    FNAME_DOUGHNUT, 
+    load_doughnut,
+    save_backup,
+    session_id_to_bids_session,
+)
+
+
+def run_dcm2bids(dicom_id, global_configs, session_id, stage, overlays, logger):
+    logger.info(f"\n***Processing participant: {dicom_id}***")
+    DATASET_ROOT = global_configs["DATASET_ROOT"]
+    SINGULARITY_PATH = global_configs["SINGULARITY_PATH"]
+    CONTAINER_STORE = global_configs["CONTAINER_STORE"]
+    DCM2BIDS_CONTAINER = global_configs["BIDS"]["dcm2bids"]["CONTAINER"]
+    DCM2BIDS_VERSION = global_configs["BIDS"]["dcm2bids"]["VERSION"]
+    DCM2BIDS_CONTAINER = DCM2BIDS_CONTAINER.format(DCM2BIDS_VERSION)
+    SINGULARITY_DCM2BIDS = f"{CONTAINER_STORE}/{DCM2BIDS_CONTAINER}"
+    SINGULARITY_WD = "/scratch"
+    SINGULARITY_DICOM_DIR = f"{SINGULARITY_WD}/dicom/ses-{session_id}"
+    SINGULARITY_BIDS_DIR = f"{SINGULARITY_WD}/bids"
+    CONFIG_FILE=f"{SINGULARITY_WD}/proc/dcm2bids_config.json"
+
+    logger.info(f"Using SINGULARITY_DCM2BIDS: {SINGULARITY_DCM2BIDS}")
+
+    flag_overlay = ''
+    flag_bind = ''
+    if overlays is not None:
+
+        logger.info(f"Using overlay(s): {overlays}")
+        for overlay in overlays:
+            flag_overlay += f'--overlay {overlay} '
+
+        for dname in ['bids', 'tabular', 'proc']:
+            flag_bind += f'--bind {DATASET_ROOT}/{dname}:{SINGULARITY_WD}/{dname} '
+    else:
+        flag_bind = f'--bind {DATASET_ROOT}:{SINGULARITY_WD}'
+
+    # Singularity CMD 
+    SINGULARITY_CMD=f"{SINGULARITY_PATH} run {flag_overlay} {flag_bind} {SINGULARITY_DCM2BIDS} "
+
+    # dcm2bids CMD
+    if stage == 1:
```

Comment:
maybe we should name the stages with more informative keyword e.g. "prepare-data", "convert-data" (or something better!)